From 5a19942c298c0b4214cffe180e4b78eed78abad9 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Thu, 14 Feb 2019 21:38:54 +0200
Subject: [PATCH 002/741] dpaa2-eth: Fix XDP rxq info registration

commit 37aea92db636b2fa241324ab008dda4bf80f34a8 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Only register XDP rxq info once per channel, regardless of
how many FQs are assigned to it.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/freescale/dpaa2/dpaa2-eth.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/freescale/dpaa2/dpaa2-eth.c b/drivers/net/ethernet/freescale/dpaa2/dpaa2-eth.c
index 60ac993ada36..296245fef96a 100644
--- a/drivers/net/ethernet/freescale/dpaa2/dpaa2-eth.c
+++ b/drivers/net/ethernet/freescale/dpaa2/dpaa2-eth.c
@@ -2510,6 +2510,9 @@ static int setup_rx_flow(struct dpaa2_eth_priv *priv,
 		return err;
 	}
 
+	if (fq->tc > 0)
+		return 0;
+
 	/* xdp_rxq setup */
 	err = xdp_rxq_info_reg(&fq->channel->xdp_rxq, priv->net_dev,
 			       fq->flowid);
-- 
2.17.1

